packages/astro/src/core/build/static-build.ts

```diff
@@ -232,7 +262,77 @@ async function clientBuild(
 	return buildResult;
 }
 
-async function cleanSsrOutput(opts: StaticBuildOptions) {
+/**
+ * For each statically prerendered page, replace their SSR file with a noop.
+ * This allows us to run the SSR build only once, but still remove dependencies for statically rendered routes.
+ */
+async function cleanStaticOutput(opts: StaticBuildOptions, internals: BuildInternals) {
+	const allStaticFiles = new Set();
+	for (const pageData of eachPrerenderedPageData(internals)) {
+		allStaticFiles.add(internals.pageToBundleMap.get(pageData.moduleSpecifier));
+	}
+	const ssr = opts.settings.config.output === 'server';
+	const out = ssr ? opts.buildConfig.server : getOutDirWithinCwd(opts.settings.config.outDir);
+	// The SSR output is all .mjs files, the client output is not.
+	const files = await glob('**/*.mjs', {
+		cwd: fileURLToPath(out),
+	});
+
+	if (files.length) {
```

Comment:
it looks like this will currently run even if the user isn't using prerendering. Could we check if `allStaticFiles.size` as well to make sure this part only runs on prerendering projects?

Reply:
I just double checked, `cleanStaticOutput` only ever runs on prerendering projects, so no need for extra logic here.